crates/red_knot_python_semantic/resources/mdtest/exception/basic.md

```diff
@@ -80,13 +80,72 @@ def foo(
 ):
     try:
         help()
-    # error: [invalid-exception-caught]
+    # error: [invalid-exception]
     except x as e:
         reveal_type(e)  # revealed: Unknown
-    # error: [invalid-exception-caught]
+    # error: [invalid-exception]
     except y as f:
         reveal_type(f)  # revealed: OSError | RuntimeError | Unknown
     except z as g:
         # TODO: should emit a diagnostic here:
         reveal_type(g)  # revealed: @Todo(full tuple[...] support)
 ```
+
+## Object raised is not an exception
+
+```py
+try:
+  raise AttributeError()  # fine
+except:
+  ...
+
+try:
+    raise FloatingPointError  # fine
+except:
+    ...
+
+try:
+    raise 1  # error: [invalid-exception]
+except:
+    ...
+
+try:
+    raise int  # error: [invalid-exception]
+except:
+    ...
+```
+
+## Exception cause is not an exception
+
+```py
+try:
+    raise EOFError() from GeneratorExit  # fine
+except:
+    ...
+
+try:
+    raise StopIteration from MemoryError()  # fine
+except:
+    ...
+
+try:
+    raise StopIteration() from None  # fine
+except:
+    ...
+
+try:
+    raise ZeroDivisionError from False  # error: [invalid-exception]
+except:
+    ...
+
+try:
+    raise SystemExit from bool()  # error: [invalid-exception]
+except:
+    ...
+
+try:
+    raise
+except int as e:  # error: [invalid-exception]
+    reveal_type(e)  # revealed: Unknown
+    raise LookupError from e
+```
```

Comment:
Might be nice to have a happy-path test similar to this but showing that we can catch an exception and `raise from` that exception without error. So like this test, but with a valid exception type in place of `int`.